txn.go

```diff
@@ -712,6 +711,7 @@ func (txn *Txn) CommitWith(cb func(error)) {
 	if cb == nil {
 		panic("Nil callback provided to CommitWith")
 	}
+	defer txn.Discard()
```

Comment:
Why two defer for the sam ething?

Reply:
I don't understand the question

Reply:
Could you rephrase it?

Reply:
Yeah, I get it, but what's the same thing? If that's about `Commit` and `CommitWith`, then it's not the same. They are two different ways of committing data and both have to Discard.

Also note that Discard protection from multiple calls, and even if a user calls both there won't be any side effects.